lib/matplotlib/figure.py

```diff
@@ -1576,6 +1576,94 @@ def subplots(self, nrows=1, ncols=1, sharex=False, sharey=False,
             # Returned axis array will be always 2-d, even if nrows=ncols=1.
             return axarr
 
+    def build_grid(self, layout, *, subplot_kw=None, gridspec_kw=None):
```

Comment:
not sure about nesting the kwargs like this.  Makes a simple thing suddenly quite complex.

Reply:
This is identical to the `subplots` API.  You need these to be able to get the subplots to be polar / 3D and to tune the paddings and such on the gridspec calls.